button-merchant/src/main/java/com/usebutton/merchant/Event.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Event.java
+ *
+ * Copyright (c) 2020 Button, Inc. (https://usebutton.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package com.usebutton.merchant;
+
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Internal class used to encapsulate event data.
+ */
+class Event {
+
+    private static final String TAG = Event.class.getSimpleName();
+
+    enum Name {
+        DEEPLINK_OPENED("btn:deeplink-opened");
+
+        private final String eventName;
+
+        Name(String eventName) {
+            this.eventName = eventName;
+        }
+
+        @Override
+        public String toString() {
+            return eventName;
+        }
+    }
+
+    enum Property {
+        URL("url");
+
+        private final String propertyName;
+
+        Property(String propertyName) {
+            this.propertyName = propertyName;
+        }
+
+        @Override
+        public String toString() {
+            return propertyName;
+        }
+    }
+
+    private final Name name;
+    @Nullable
+    private final String sourceToken;
```

Comment:
I named this `attributionToken` for consistency with the current naming throughout the ML

Reply:
The Android ML refers to it (internally only) as sourceToken. Maybe we can get parity in a later PR.